File: tslib-py/tslib/moving_average.py
from typing import List, Optional

from ._http import _Session


class MovingAverageAPI:
    def __init__(self, session: _Session) -> None:
        self._s = session

    def compute(
        self,
        data: List[float],
        *,
        period: int = 5,
        type: str = "SIMPLE",
        alpha: Optional[float] = None,
    ) -> List[float]:
        body: dict = {"data": data, "period": period, "type": type}
        if alpha is not None:
            body["alpha"] = alpha
        return list(self._s.post("/api/moving-average", body)["forecasts"])

File: tslibR/R/moving_average.R
#' Compute a moving average
#'
#' @param client A [`tslib_client`] object.
#' @param data Numeric vector of observations.
#' @param period Window size (default 5).
#' @param type Moving average type: `"SIMPLE"` (SMA), `"EMA"` (exponential),
#'   `"WMA"` (weighted), or `"CMA"` (cumulative). Default `"SIMPLE"`.
#' @param alpha Smoothing factor for EMA (optional; server uses a default when
#'   omitted).
#' @return Numeric vector of moving-average values.
#' @export
tslib_moving_average <- function(client, data,
                                 period = 5L,
                                 type   = "SIMPLE",
                                 alpha  = NULL) {
  body <- .compact(list(
    data   = as.list(data),
    period = period,
    type   = type,
    alpha  = alpha
  ))
  resp <- .tslib_post(client, "/api/moving-average", body)
  unlist(resp[["forecasts"]])
}
